presto-main/src/main/java/com/facebook/presto/execution/CreateTableTask.java

```diff
@@ -72,12 +89,37 @@ public String explain(CreateTable statement)
         }
 
         List<ColumnMetadata> columns = new ArrayList<>();
+        Map<String, Object> finalProperties = new HashMap<>();
         for (TableElement element : statement.getElements()) {
-            Type type = metadata.getType(parseTypeSignature(element.getType()));
-            if ((type == null) || type.equals(UNKNOWN)) {
-                throw new SemanticException(TYPE_MISMATCH, element, "Unknown type for column '%s' ", element.getName());
+            if (element instanceof ColumnDefinition) {
+                ColumnDefinition column = (ColumnDefinition) element;
+                Type type = metadata.getType(parseTypeSignature(column.getType()));
+                if ((type == null) || type.equals(UNKNOWN)) {
+                    throw new SemanticException(TYPE_MISMATCH, column, "Unknown type for column '%s' ", column.getName());
+                }
+                columns.add(new ColumnMetadata(column.getName(), type));
+            }
+            else if (element instanceof LikeClause) {
+                LikeClause likeClause = (LikeClause) element;
+                QualifiedObjectName likeTableName = createQualifiedObjectName(session, statement, likeClause.getTableName());
+                Optional<TableHandle> likeTable = metadata.getTableHandle(session, likeTableName);
+                if (!likeTable.isPresent()) {
+                    throw new SemanticException(MISSING_TABLE, statement, "Like table '%s' doesn't exist", likeTableName);
+                }
+
+                TableMetadata likeTableMetadata = metadata.getTableMetadata(session, likeTable.get());
+
+                Map<String, Object> currentProperties = likeTableMetadata.getMetadata().getProperties();
+                mergeTableProperties(currentProperties, finalProperties, likeClause.getLikeOptions());
+
+                List<ColumnMetadata> likeColumns = likeTableMetadata.getVisibleColumns();
+                for (ColumnMetadata columnMetadata : likeColumns) {
+                    columns.add(new ColumnMetadata(columnMetadata.getName(), columnMetadata.getType()));
+                }
+            }
+            else {
+                throw new PrestoException(StandardErrorCode.NOT_SUPPORTED, "Unsupported TableElement: " + element);
```

Comment:
This should be `IllegalArgumentException` or `INTERNAL_ERROR` since it is a programming bug